Python/fileutils.c

```diff
@@ -1344,9 +1348,25 @@ _Py_wfopen(const wchar_t *path, const wchar_t *mode)
     if (cpath == NULL) {
         return NULL;
     }
+    if (_Py_IsCoreInitialized() &&
+        PySys_Audit("open", "uui", path, mode, 0) < 0) {
```

Comment:
I'm not sure that the GIL is held to call this function.

Reply:
I've moved the auditing for `_Py_wfopen` and `_Py_fopen` to the call sites, since most of them do actually hold the GIL.

Reply:
I see many calls in Modules/getpath.c and PC/getpathp.c. In these files, there is no interpreter yet nor GIL, no? I only see one call in pymain_run_file() which should hold the GIL and have an intererpreter running.

Reply:
Well, PySys_Audit() should detect if there is an interpreter or not. It is just a remark :-)

Reply:
You're right, if DTrace is enabled then we'll get past the other checks. I'll fix that up.

Hopefully in the future we can reenable primitive types outside of a specific interpreter and then won't have to skip early events (some of us are already thinking about this, but it's not going to happen right now).

Reply:
And I'll put the audit call back into these functions, since it'll have no effect when the GIL is not held (until the point where it can have an effect)